test/modules/carto/api/maps-api-client.spec.js

```
@@ -1,101 +1,294 @@
+      json: () => TILEJSON_RESPONSE,
+      ok: true
+    });
+  };
+
+  let data = await _getDataV1({
+    type: MAP_TYPES.QUERY,
+    source: 'select * from a',
+    credentials: getDefaultCredentials()
+  });
+
+  t.ok(
+    Array.isArray(data.tiles) && data.tiles.length === 1,
+    'tiles should be an array with content'
+  );
 
-    if (!tc.server.ok) {
-      setDefaultCredentials(tc.credentials);
+  _global.fetch = url => {
+    t.is(
+      url,
+      'https://maps-v2/bigquery/tileset?source=tileset&format=tilejson&api_key=default_public&client=deck-gl-carto',
+      'should be a right url for a tileset at v2'
+    );
+    return Promise.resolve({
+      json: () => TILEJSON_RESPONSE,
+      ok: true
+    });
+  };
+
+  data = await _getDataV1({
+    type: MAP_TYPES.TILESET,
+    source: 'tileset',
+    credentials: getDefaultCredentials()
+  });
 
-      try {
-        await _getTileJSON(CFG);
-      } catch (err) {
-        t.pass(`Correctly throws ${err}`);
+  t.ok(
+    Array.isArray(data.tiles) && data.tiles.length === 1,
+    'tiles should be an array with content'
+  );
+
+  _global.fetch = fetch;
+  setDefaultCredentials({});
+  t.end();
+});
+
+test('getData#parameters', async t => {
+  const params = {
+    type: MAP_TYPES.TABLE,
+    connection: 'connection_name',
+    source: 'table'
+  };
+
+  try {
+    await getData({
+      ...params,
+      credentials: {
+        apiVersion: API_VERSIONS.V2
       }
-      /* eslint-disable-next-line no-continue */
-      continue;
-    }
+    });
+    t.fail('it should throws an error');
+  } catch (e) {
+    t.is(e.message, 'Method only available for v3', 'should throw when no v3');
+  }
 
-    const func = await _getTileJSON(CFG);
-    t.ok(func.tiles[0] === SQL_RESPONSE.tiles[0], 'getTileJSON correctly returns tiles endpoint');
+  try {
+    await getData({
+      ...params,
+      credentials: {
+        apiVersion: API_VERSIONS.V3
+      }
+    });
+    t.fail('it should throws an error');
```

typo --> "it should throw an error"